File: src/Troop.cpp
#include "Troop.h"

Troop::Troop(double base_atk, double base_def, double base_hp, TroopType type) {
    this->base_atk = base_atk;
    this->base_def = base_def;
    this->base_hp = base_hp;
    this->type = type;
}

double Troop::getAtk() { return this->base_atk; }
double Troop::getDef() { return this->base_def; }
double Troop::getHp() { return this->base_hp; }
TroopType Troop::getType() { return this->type; }

File: src/main/java/shrunken/rokcc/sim/Troop.java
package shrunken.rokcc.sim;

public class Troop {
    private double atk;
    private double def;
    private double hp;
    private TroopType troopType;

    public Troop(double atk, double def, double hp, TroopType troopType) {
        this.atk = atk;
        this.def = def;
        this.hp = hp;
        this.troopType = troopType;
    }

    public double getAtk() {return this.atk;}
    public double getDef() {return this.def;}
    public double getHp() {return this.hp;}
    public TroopType getTroopType() {return this.troopType;}
}
